utils/utils_test.go

```diff
@@ -0,0 +1,29 @@
+package utils
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRandomPick(t *testing.T) {
+	testCases := []struct {
+		desc              string
+		inPickProbability float64
+		expected          bool
+	}{
+		{
+			desc:              "Zero logging rate. Expect false",
+			inPickProbability: 0.00,
+			expected:          false,
+		},
+		{
+			desc:              "100% logging rate, expect true",
+			inPickProbability: 1.00,
+			expected:          true,
+		},
+	}
+	for _, tc := range testCases {
+		assert.Equal(t, tc.expected, RandomPick(tc.inPickProbability), tc.desc)
```

Comment:
Can you use `t.Run` here too? .. and also change desc to name. Test names can be simplified to "zero" and "one". Thank you for not trying to test the random number generator part. :)

Reply:
Sure. Modified

Reply:
Do you agree with naming this function `RandomPick`? I'm not so sure about the name but, I can't think of anything better of the top of my head.

Reply:
It doesn't bother me. Other options to consider:
- `ShouldSample`
- `ShouldLog`
- `RandomChance`